contracts/src/plugin/setup/PluginUpgradeableSetup.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity ^0.8.8;
+
+import {ERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+
+import {IProtocolVersion} from "../../utils/versioning/IProtocolVersion.sol";
+import {ProtocolVersion} from "../../utils/versioning/ProtocolVersion.sol";
+import {IPluginSetup} from "./IPluginSetup.sol";
+
+/// @title PluginUpgradeableSetup
+/// @author Aragon Association - 2022-2024
+/// @notice An abstract contract to inherit from to implement the plugin setup for upgradeable plugins, i.e, `PluginUUPSUpgradeable` being deployed via the UUPS pattern (see [ERC-1822](https://eips.ethereum.org/EIPS/eip-1822) and [ERC-1967](https://eips.ethereum.org/EIPS/eip-1967)).
+/// @custom:security-contact sirt@aragon.org
+abstract contract PluginUpgradeableSetup is ERC165, IPluginSetup, ProtocolVersion {
+    /// @notice The address of the plugin implementation contract for initial block explorer verification
+    /// and to create [ERC-1967](https://eips.ethereum.org/EIPS/eip-1967) UUPS proxies from.
+    address internal immutable IMPLEMENTATION;
+
+    /// @notice Thrown when an update path is not available, for example, if this is the initial build.
+    /// @param fromBuild The build number to update from.
+    /// @param thisBuild The build number of this setup to update to.
+    error InvalidUpdatePath(uint16 fromBuild, uint16 thisBuild);
+
+    /// @notice The contract constructor, that setting the plugin implementation contract.
+    /// @param _implementation The address of the plugin implementation contract.
+    constructor(address _implementation) {
+        IMPLEMENTATION = _implementation;
+    }
+
+    /// @inheritdoc IPluginSetup
+    /// @dev The default implementation for the initial build 1 that reverts because no earlier build exists.
+    function prepareUpdate(
+        address _dao,
+        uint16 _fromBuild,
+        SetupPayload calldata _payload
+    ) external pure virtual returns (bytes memory, PreparedSetupData memory) {
+        (_dao, _fromBuild, _payload);
+        revert InvalidUpdatePath({fromBuild: 0, thisBuild: 1});
```

Comment:
as discussed I am not a fan of this error.
Imaging the following:
- PluginRepo gets created with placeholder versions (build 1 and build 2)
- Dev doesn't override this function with the initial build 3

Then the error is wrong because `fromBuild` and `thisBuild` report the wrong build numbers.
As suggested in our call I would rather have an empty function or no implementation at all to force the dev to implement it.